### PR TITLE
Add GitHub workflow to automatically tag each commit on `main`

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: .github/workflows/permaref.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  push:
+    branches:
+      - "main"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (BurntSushi) [inline .github/workflows/permaref.yaml]: I think v4 is out

Comment (zanieb) [reply .github/workflows/permaref.yaml]: ty

File: .github/workflows/permaref.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Get the permanent ref number
+        id: get_version
+        run: |
+          # Enable pipefail so git command failures do not result in null versions downstream
+          set -x
+
+          echo ::set-output name=LAST_PERMA_NUMBER::$(\
+            git ls-remote --tags --refs --sort="v:refname" \
+            https://github.com/zanieb/pubgrub.git | grep "tags/perma-" | tail -n1 | sed 's/.*\/perma-//' \
+          )
```
Comment (BurntSushi) [inline .github/workflows/permaref.yaml:27]: I think you can do the slightly nicer `echo {name}={value} >> "$GITHUB_OUTPUT"` now.

Comment (zanieb) [reply .github/workflows/permaref.yaml:27]: Yeah good note I was about to comment here that I should change to the new syntax.

I copied this from some code I wrote a long time ago to pull versions from tags.

Comment (BurntSushi) [reply .github/workflows/permaref.yaml:27]: > I copied this from some code I wrote a long time ago to pull versions from tags.

We have the exact same GHA development workflow. :P

Comment (zanieb) [reply .github/workflows/permaref.yaml:27]: #3

**Pull request merged**